dali/core/mm/cu_vm_test.cc

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) 2021, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/core/mm/cu_vm.h"
+#include <gtest/gtest.h>
+
+#if DALI_USE_CUDA_VM_MAP
+
+namespace dali {
+namespace mm {
+namespace test {
+
+TEST(CUMemAddressRange, Reserve) {
+  ASSERT_TRUE(cuInitChecked());
+  int64_t requested = 4000000;
+  cuvm::CUMemAddressRange range = cuvm::CUMemAddressRange::Reserve(requested);
+  EXPECT_GE(static_cast<int64_t>(range.size()), requested);
+  CUpointer_attribute attrs[3] = {
+    CU_POINTER_ATTRIBUTE_RANGE_START_ADDR,
+    CU_POINTER_ATTRIBUTE_RANGE_SIZE,
+    CU_POINTER_ATTRIBUTE_MAPPED
+  };
+  CUdeviceptr start = 0;
+  size_t size = 0;
+  bool mapped = false;
+  void *data[3] = { &start, &size, &mapped };
+  CUDA_CALL(cuPointerGetAttributes(3, attrs, data, range.ptr() + 100));
```

Comment:
Why you take the address from somewhere inside the range?

Reply:
Because I want to check that the VA machinery worked properly and I'm not getting the base address by sheer coincidence.

Reply:
ok